lineapy/graph_reader/artifact_collection.py

```diff
@@ -346,3 +353,75 @@ def get_session_module(self, dependencies: TaskGraphEdge = {}):
             return session_module
 
         return
+
+    def generate_pipeline_files(
```

Comment:
this looks like it belongs to writers. my basic criteria here is artifact_collection - responsible for knowing what is where and how its linked to each other. doing jinjatemplate.render - responsibility of writers. artifact_collection should not know what template to call and what parameters to pass how. it just holds the relationship and maybe an instances of different writers (prefer lazy loading but this is a perf change that can come later.).

Reply:
Moved it to the pipeline_writers.py .